src/Hadolint/Rules.hs

```diff
@@ -546,7 +547,23 @@ useAdd = instructionRule code severity message check
             , format <- archiveFormats
             ]
     check _ = True
-    archiveFormats = [".tar", ".gz", ".bz2", "xz"]
+    archiveFormats =
+      [ ".tar"
+      , ".tar.bz2"
+      , ".tb2"
+      , ".tbz"
+      , ".tbz2"
+      , ".tar.gz"
+      , ".tgz"
+      , ".tpz"
+      , ".tar.lz"
+      , ".tar.lzma"
+      , ".tlz"
+      , ".tar.xz"
+      , ".txz"
+      , ".tar.Z"
+      , ".tZ"
```

Comment:
What about zip files?

Reply:
do you know if `ADD` is able to extract zip files? I cannot remember

Reply:
No, it does not. 

Ref.:
````
If <src> is a local tar archive in a recognized compression format (identity, gzip, bzip2 or xz) then it is unpacked as a directory. Resources from remote URLs are not decompressed. When a directory is copied or unpacked, it has the same behavior as tar -x, the result is the union of:

Whatever existed at the destination path and
The contents of the source tree, with conflicts resolved in favor of “2.” on a file-by-file basis.
````
Source https://docs.docker.com/engine/reference/builder/#add